src/components/Fund/FundDetails/FundDetailsContainer.js

```diff
@@ -53,94 +54,108 @@ const FundDetailsContainer = ({
   mutator,
   onClose,
   onEdit,
-  resources,
   stripes,
 }) => {
-  useEffect(() => {
-    mutator.fund.reset();
-    mutator.fundType.reset();
-    mutator.ledger.reset();
-    mutator.allocatedFrom.reset();
-    mutator.allocatedTo.reset();
-    mutator.group.reset();
-    mutator.budgets.reset();
+  const [fetchFundResource, setFetchFundResource] = useState(false);
+  const [compositeFund, setCompositeFund] = useState({ fund: {}, groupIds: [] });
+  const [fundTypeName, setFundTypeName] = useState('');
+  const [ledger, setLedger] = useState({});
+  const [allocatedFromFunds, setAllocatedFromFunds] = useState('');
+  const [allocatedToFunds, setAllocatedToFunds] = useState('');
+  const [budgets, setBudgets] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
 
-    mutator.fund.GET().then(response => {
-      const {
-        fund: {
-          ledgerId,
-          fundTypeId,
-          allocatedFromIds,
-          allocatedToIds,
-        },
-      } = response;
-
-      if (fundTypeId) {
-        mutator.fundType.GET({
-          params: {
-            query: `id==${fundTypeId}`,
+  const fetchFundDetails = useCallback(
+    () => {
+      setIsLoading(true);
+      mutator.fund.GET().then(response => {
+        setCompositeFund(response);
+        const {
+          fund: {
+            ledgerId,
+            fundTypeId,
+            allocatedFromIds,
+            allocatedToIds,
           },
-        });
-      }
+        } = response;
 
-      if (ledgerId) {
-        mutator.ledger.GET({
-          params: {
-            query: `id==${ledgerId}`,
-          },
-        });
-      }
+        if (fundTypeId) {
+          mutator.fundType.GET({
+            params: {
+              query: `id==${fundTypeId}`,
```

Comment:
what is the reason to use GET on list here and for other things around, if you have a specific id? IMHO, GET by id is more appropriate api, what do you think?